### PR TITLE
(feat)payments-server: update create subscription page for paypal designs

### PR DESCRIPTION
**Pull request by birdsarah**
## Because

- The create subscription screen did not match the user tested designs

## This pull request

- adds centered headings
- reorganizes form code, now that payment is more complicated
- updates the paypal button
- updates the privacy policy links
- moves the test for legal text to the createSubscription tests

## Issue that this pull request solves

Closes: #7449

## Checklist

_Put an `x` in the boxes that apply_

- [x] My commit is GPG signed.
- [x] If applicable, I have modified or added tests which pass locally.
- [ ] I have added necessary documentation (if appropriate).
- [ ] I have verified that my changes render correctly in RTL (if appropriate).

## Screenshots 

Before:
Screenshot from 2021-02-11 20-13-12

After:
Screenshot from 2021-02-11 20-08-48

Design should roughly be [panel 288 from here](https://www.figma.com/file/AvdKdmts8KN8HK6d1VCMTH/Paypal_Checkout_Concept(B)?node-id=0%3A1) - jess cook can provide more guidance.

## Other information 

Note that the paypal button is delivered automatically from paypal. despite the docs (https://developer.paypal.com/docs/business/checkout/reference/style-guide/#size) it does not want to give us a button with a P logo in it. My current best guess for this is that the buttons have been updated but the docs haven't (see "update your buttons" here - https://www.paypal.com/us/webapps/mpp/logos-buttons and note we're doing "Checkout")

### CODE REVIEW

File: packages/fxa-payments-server/src/components/PaymentLegalBlurb/index.tsx [hunk 1]
```diff
@@ -5,22 +5,32 @@ import './index.scss';
...
+            href="https://paypal.com/privacy"
+            target="_blank"
+            rel="noopener noreferrer"
+          ></a>
+        ),
       }}
     >
       <p>
-        View the <a>Stripe privacy policy</a>.
+        View the <stripe>Stripe privacy policy</stripe> and{' '}
```
Comment (birdsarah) [inline packages/fxa-payments-server/src/components/PaymentLegalBlurb/index.tsx:32]: this made react unhappy but i wasn't sure how to ref the elems - guidance appreciated.

File: packages/fxa-payments-server/src/routes/Product/SubscriptionCreate/index.scss [hunk 2]
```diff
@@ -20,16 +20,43 @@
...
+
+  .subscription-create-pay-with-other .paypal-button {
+    width: 120px;
+    height: 35px;
+    margin: auto; 
+    position: relative;
+      div {
+        // This is very hacky, but without it the button is off-center
+        position: absolute;
+        right: -8px;
```
Comment (birdsarah) [inline packages/fxa-payments-server/src/routes/Product/SubscriptionCreate/index.scss:47]: this is gross but something like it is necessary because otherwise the button looks just slightly off center - presumably because it's actually an iframe inside this div.

**Comment by jess-cook03** [comment]
@birdsarah 
I saw your note about the paypal button/code and styling.... this is very interesting.
I wonder how other platforms/sites/companies are able to leverage the other button styles from paypal then? 
Screen Shot 2021-02-12 at 12 08 52 PM
Screen Shot 2021-02-12 at 12 06 57 PM

Just thinking examples I have seen whilst shopping online. Does this help at all -- https://developer.paypal.com/docs/checkout/integration-features/customize-button/#color ?

- **Is it possible to increase the size of the paypal button?** Looks pretty tiny. Height I would increase to match the height of the pay now button below (Button L, 48px) ... and length increase to stretch wider and fill up a bit more of the section as well as have it stand out in comparison to the stripe 'pay now' button blow. 

- The padding between the header line/text and button -- can we increase the padding ratios here to achieve more vertical rhythm? Right now its all looking a bit squished together. See image below for suggested units of 8px increased padding:
Screen Shot 2021-02-12 at 12 32 20 PM


I hope this is the kind of feedback you were looking for?
I know this is not yet pixel perfect... so hopefully this isn't too nitty gritty too soon.
Ping me if you have questions!

**Comment by birdsarah** [comment]
> If we need to change the button shape to be square with slightly rounded corners (like these examples) in order to pull this code for that kind of button from paypal then thats fine too. We don't have to do the pill shape if that is causing us issues.

It's not the pill shape I couldn't figure out a variation from the style options that I had. I was using these docs (https://developer.paypal.com/docs/business/checkout/reference/style-guide/) which appear to the the same as docs you linked (https://developer.paypal.com/docs/checkout/integration-features/customize-button/#color) but under a different url. sigh.

I looked up your baggu option which is a shopify site.  Their div that corresponds to ours is:

```
<div id="zoid-paypal-button-1790a1159d" class="paypal-button paypal-button-context-iframe paypal-button-label-paypal paypal-button-size-responsive paypal-button-layout-horizontal" data-paypal-smart-button-version="4.0.325" style="height: 42px;"><style>
```

Ours is 

```
<div id="zoid-paypal-buttons-06b233dbc8_mtc6ndu6mdm" class="paypal-buttons paypal-buttons-context-iframe paypal-buttons-label-paypal paypal-buttons-layout-horizontal" data-paypal-smart-button-version="5.0.201" style="height: 35px; transition: all 0.2s ease-in-out 0s;"><style>
```

Note that their button version is 4.x and ours is 5.x. I don't know what to tell you. I really don't think we have control over this. Sorry :(

Your smallwood example is also a shopify site with the same 4.x version of paypal. 

If you can find a non-shopify site, let me know and I'll take a look at that code. 

Perhaps @bbangert or @bdanforth can weigh in.

====


>  Is it possible to increase the size of the paypal button? Looks pretty tiny. Height I would increase to match the height of the pay now button below (Button L, 48px) ... and length increase to stretch wider and fill up a bit more of the section as well as have it stand out in comparison to the stripe 'pay now' button blow.

Yes - can be whatever size you want. 48 high works. You give me a width

> The padding between the header line/text and button -- can we increase the padding ratios here to achieve more vertical rhythm? Right now its all looking a bit squished together. See image below for suggested units of 8px increased padding

I can't be quite as precise but I can get close.

**Comment by birdsarah** [comment]
@jess-cook03 I've made the pixel changes you described. Should be spot on to your specs. See screenshots below. 

This PR still needs an assist from @chenba or someone to help with my elems Localized fail. But otherwise I think it's good to go.

Screenshot from 2021-02-13 13-57-20

Screenshot from 2021-02-13 13-57-27

**Comment by jess-cook03** [comment]
@birdsarah The pixel changes do look good! Thank you for this update! 👍

File: packages/fxa-payments-server/src/routes/Product/SubscriptionCreate/index.scss [hunk 3]
```diff
@@ -20,16 +20,38 @@
...
 
   .billing-title {
     font-size: 15px;
     font-weight: 600;
     margin-bottom: 20px;
-
-    @include min-width('tablet') {
-      border-top: 1px solid #e0e0e6;
-      padding-top: 16px;
-    }
```
Comment (chenba) [inline packages/fxa-payments-server/src/routes/Product/SubscriptionCreate/index.scss:32]: Why was this removed?

**Comment by birdsarah** [comment]
> @birdsarah silly question perhaps.... but are the fonts that are being used here Inter for majority (body/labels) and Metropolis for larger headings...? From these screenshots the fonts look different, which could just be the dev environment... anyways just wanted to ask!

@jess-cook03 - I haven't done anything to change fonts. The fonts are consistent on the page. For text I've added e.g. a <p> tag. It's just using what <p> tag already is.

File: packages/fxa-payments-server/src/routes/Product/SubscriptionCreate/index.scss (hunk 3 above)
Comment (birdsarah) [reply packages/fxa-payments-server/src/routes/Product/SubscriptionCreate/index.scss:32]: whimsy??? jk - my bad - thanks for the catch.

**Review by chenba — APPROVED** [review]

**Comment by birdsarah** [comment]
Oooohhh - I just found a bug - this looks suboptimal if paypal doesn't load. going to correct.

File: packages/fxa-payments-server/src/routes/Product/SubscriptionCreate/index.scss (hunk 3 above)
Comment (birdsarah) [reply packages/fxa-payments-server/src/routes/Product/SubscriptionCreate/index.scss]: HAH - I just remembered why I removed this. We don't want the horizontal border it adds anymore.

Because "Billing Information" text and card images order have been changed. The addition of the horizontal lines is now different.

Screenshot from 2021-02-16 12-06-51

This is what it looks like with that border in. Now removing it again.

**Comment by birdsarah** [comment]
OK. Made sure we've got three scenarios covered:
1) FXA using paypal no existing subscription - should say "Or pay with card"
2) FXA not using paypal no existing subscription - should say "Pay with card"
3) Has existing card on file - should display billing information.

Screenshots below

(1) "Or pay with card"
Screenshot from 2021-02-16 13-26-48

(2) "Pay with card"
Screenshot from 2021-02-16 13-58-58


(3) "Card on file"
Screenshot from 2021-02-16 13-27-34

**Comment by birdsarah** [comment]
Got a LGTM from @jess-cook03

File: packages/fxa-payments-server/src/routes/Product/SubscriptionCreate/index.tsx [hunk 4]
```diff
@@ -180,51 +182,80 @@ export const SubscriptionCreate = ({
...
+                </div>
+              </Suspense>
+            )}
+          </div>
 
-          {!hasExistingCard(customer) && <AcceptedCards />}
+          <div className="subscription-create-pay-with-card">
+            {!hasExistingCard(customer) && !paypalScriptLoaded && (
+              <div>
+                <Localized id="pay-with-heading-card-only">
```
Comment (chenba) [inline packages/fxa-payments-server/src/routes/Product/SubscriptionCreate/index.tsx:208]: `pay-with-heading-card-only` and `pay-with-heading-card-or` are missing in the .ftl

File: packages/fxa-payments-server/src/routes/Product/SubscriptionCreate/index.tsx [hunk 5]
```diff
@@ -180,51 +182,80 @@ export const SubscriptionCreate = ({
...
+                    currencyCode={selectedPlan.currency}
+                  />
+                </div>
+              </Suspense>
+            )}
+          </div>
 
-          {!hasExistingCard(customer) && <AcceptedCards />}
+          <div className="subscription-create-pay-with-card">
+            {!hasExistingCard(customer) && !paypalScriptLoaded && (
```
Comment (chenba) [inline packages/fxa-payments-server/src/routes/Product/SubscriptionCreate/index.tsx:206]: If the customer has an existing subscription paid with PayPal but the script failed to load for whatever reason, they'd get the CC form here I think.

**Review by chenba — CHANGES_REQUESTED** [review]
Looks good overall.  Some missing/mis-matchig l10n messages.  There's a potential error condition where a subscriber paying with PayPal end up seeing the Stripe CC form.

File: packages/fxa-payments-server/src/routes/Product/SubscriptionCreate/index.tsx (hunk 4 above)
Comment (birdsarah) [reply packages/fxa-payments-server/src/routes/Product/SubscriptionCreate/index.tsx:208]: Resolved.

**Comment by birdsarah** [comment]
If they have an existing paypal subscription, and end up back on this page it looks like this:

Screenshot from 2021-02-16 20-23-26

This is rough, but @lesleyjanenorton has a seperate PR that provides a helper method to let us know what the payment provider is and we can then clean this up. @lesleyjanenorton I'm not sure whether you want to include fixing this in your PR (probably not) or open a new issue.

File: packages/fxa-payments-server/src/routes/Product/SubscriptionCreate/index.tsx (hunk 5 above)
Comment (birdsarah) [reply packages/fxa-payments-server/src/routes/Product/SubscriptionCreate/index.tsx:206]: No, this block does not control the display of the CC form. That's handled in Payment Form.

I have tried to confirm this by verifying this scenario locally and you're presented with the screen as shown below (with the awkward "last4") as an existing PayPal account consists.

To confirm I tried both disabling paypal between loads and updating the paypalScriptLoaded variable.

Comment (chenba) [reply packages/fxa-payments-server/src/routes/Product/SubscriptionCreate/index.tsx:206]: > No, this block does not control the display of the CC form. ...

Oh right, duh 🤦

File: packages/fxa-payments-server/src/components/PaymentLegalBlurb/index.tsx [hunk 6]
```diff
@@ -3,25 +3,36 @@ import { Localized } from '@fluent/react';
...
+            target="_blank"
+            rel="noopener noreferrer"
+          ></a>
+        ),
       }}
     >
-      <p>
-        View the <a>Stripe privacy policy</a>.
-      </p>
+      <p>{getPrivacyLinkText()}</p>
```
Comment (chenba) [inline packages/fxa-payments-server/src/components/PaymentLegalBlurb/index.tsx:35]: Sorry, I just realized this annoying thing today, and it might be a Product question: we probably do not want to show any PayPal related content unless we have the feature flag on.

**Review by chenba — APPROVED** [review]
👍  But might need to hide PayPal related content behind the feature flag.

File: packages/fxa-payments-server/src/components/PaymentLegalBlurb/index.tsx (hunk 6 above)
Comment (birdsarah) [reply packages/fxa-payments-server/src/components/PaymentLegalBlurb/index.tsx:35]: I added https://jira.mozilla.com/browse/FXA-3061 for this.

**Comment by birdsarah** [comment]
I'm going to finish this now lesley's pr is merged.
